File: usi/Injector.java
package injector;

import java.net.InetAddress;
import java.net.Socket;

public class Injector {
    public static final String BEGIN = "{firstname:'maurice',lastname:'plutanfiard',mail:'maurice.plutanfiard";
    public static final String END = "@gmail.com',password:'xxxxx'}";
    public static final String CONTENT_TYPE = "Content-Type: application/json\r\n\r\n";
    public static final String POST = "POST ";
    public static final String LENGTH = " HTTP/1.0\r\nHost: localhost\r\nContent-Length: ";
    public static final String JUMP = "\r\n";
    public static final String HOSTNAME = "localhost";
    public static final String URI = "/api/user";
    public static final int PORT = 8080;

    public static void main(String[] args) {
        try {
            InetAddress addr = InetAddress.getByName(HOSTNAME);
            Socket socket;
            StringBuilder builder;
            for (int i = 0; i < 1000; i++) {
                socket = new Socket(addr, PORT);
                builder = new StringBuilder();
                builder.append(POST);
                builder.append(URI);
                builder.append(LENGTH);
                builder.append(BEGIN.length() + END.length() + String.valueOf(i).length() + 4 );
                builder.append(JUMP);
                builder.append(CONTENT_TYPE);
                builder.append(JUMP);
                builder.append(JUMP);
                builder.append(BEGIN);
                builder.append(String.valueOf(i));
                builder.append(END);
                socket.getOutputStream().write(builder.toString().getBytes());
                socket.getOutputStream().flush();
                socket.getInputStream().close();
                socket.close();
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: usi/injector.c
#include <arpa/inet.h>
#include <assert.h>
#include <errno.h>
#include <netinet/in.h>
#include <signal.h>
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <sys/types.h>
#include <sys/socket.h>
#include <sys/wait.h>
#include <netdb.h>
#include <unistd.h>

#define SA      struct sockaddr
#define MAXLINE 4096
#define MAXSUB  400


#define LISTENQ         1024

extern int h_errno;

ssize_t process_http(int sockfd, char *host, char *page, char *poststr) {
	char sendline[MAXLINE + 1], recvline[MAXLINE + 1];
	ssize_t n;
	snprintf(sendline, MAXSUB,
		 "POST %s HTTP/1.0\r\n"
		 "Host: %s\r\n"
		 "Content-type: application/json\r\n"
		 "Content-length: %d\r\n\r\n"
		 "%s", page, host, strlen(poststr), poststr);
	write(sockfd, sendline, strlen(sendline));
	while ((n = read(sockfd, recvline, MAXLINE)) > 0) {

		recvline[n] = '\0';
		//printf("%s", recvline);
	}
	return n;
}

int main(void) {
	int sockfd;
	struct sockaddr_in servaddr;

	char **pptr;
	//********** You can change. Puy any values here *******
	char *hname = "soap-bubble";
	char *page = "/api/user";
	char *poststr = "{firstname:'maurice',lastname:'plutanfiard',mail:'maurice.plutanfiard@gmail.com',password:'xxxxx'}\r\n";
	//*******************************************************
	char str[50];
	char url[MAXLINE + 1];
	struct hostent *hptr;
	if ((hptr = gethostbyname(hname)) == NULL) {
		fprintf(stderr, " gethostbyname error for host: %s: %s",
			hname, hstrerror(h_errno));
		exit(1);
	}
	printf("hostname: %s\n", hptr->h_name);
	if (hptr->h_addrtype == AF_INET
	    && (pptr = hptr->h_addr_list) != NULL) {
		printf("address: %s\n",
		       inet_ntop(hptr->h_addrtype, *pptr, str,
				 sizeof(str)));
	} else {
		fprintf(stderr, "Error call inet_ntop \n");
	}
	int i = 0, count = 510000;
	for (i=500001; i<1000000; i++) {
		sockfd = socket(AF_INET, SOCK_STREAM, 0);
		bzero(&servaddr, sizeof(servaddr));
		servaddr.sin_family = AF_INET;
		servaddr.sin_port = htons(8080);
		inet_pton(AF_INET, str, &servaddr.sin_addr);
		connect(sockfd, (SA *) & servaddr, sizeof(servaddr));
		if (i == count) {
			printf("call : %d\n", i);
			count = count + 10000;
		}
		sprintf(url, "{firstname:'maurice',lastname:'plutanfiard',mail:'maurice.plutanfiard%i@gmail.com',password:'xxxxx'}\r\n", i);
		//printf("%s", url);
		process_http(sockfd, hname, page, url);
		close(sockfd);
	}
	exit(0);
}
